src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1434,6 +1444,69 @@ public int executeHashNotCalculated(Rope rope) {
         }
     }
 
+    @NodeChildren({
+            @NodeChild(type = RubyNode.class, value = "encoding"),
+            @NodeChild(type = RubyNode.class, value = "bytes"),
+            @NodeChild(type = RubyNode.class, value = "start"),
+            @NodeChild(type = RubyNode.class, value = "end")
+    })
+    public abstract static class EncodingLengthNode extends RubyNode {
+
+        public static EncodingLengthNode create() {
+            return RopeNodesFactory.EncodingLengthNodeGen.create(null, null, null, null);
+        }
+
+        public abstract int executeLength(Encoding encoding, byte[] bytes, int start, int end);
+
+        @Specialization(guards = "encoding.isUTF8()")
+        int utf8Length(Encoding encoding, byte[] bytes, int start, int end) {
+            return UTF8Encoding.class.cast(encoding).length(bytes, start, end);
```

Comment:
Why not `((UTF8Encoding) encoding)` ?

Reply:
The call is still virtual in that case.

Reply:
There are no subclasses of UTF8Encoding, why would it still be virtual?

Reply:
That's also a lot of unprofiled code, I'm not sure it's advantageous to PE it.

Reply:
I guess I'm mistaken. I thought I saw it still make a virtual call, but I must be confusing it with something else. I think using `UTF8Encoding.class.cast` is still advantageous because it avoids the IDE warning about a redundant cast (at least in IntelliJ).

Reply:
As for PE, the generated graph didn't look crazy to me. But if it is truly a problem, this node now provides a simple way to swap out the implementation and pick up those advantages throughout.

Reply:
Mmh, I think the direct cast would be clearer. (Can the IDE warning be suppressed easily with `@SuppressWarnings`?)
OK, it just seemed like a lot of branches and early return.